File: Programming-languages-and-laboratory/Java/Exercises/36_compito_java_2020_01_16/Supermercato.java
public class Supermercato {
	protected int codice;
	protected String tipo;
	protected String indirizzo;
	protected int numeroVisite;

	public Supermercato(int codice, String tipo, String indirizzo) {
		this.codice = codice;
		this.tipo = tipo;
		this.indirizzo = indirizzo;
		numeroVisite = 0;
	}
	
}

File: Programming-languages-and-laboratory/Python/Exercises/36_compito_python_20200116/Supermercato.py
class Supermercato:
    def __init__(self, codice, tipo, indirizzo, superficie, addetti):
        self._codice = codice
        self._tipo = tipo
        self._indirizzo = indirizzo
        self._superficie = superficie
        self._addetti = addetti
        self._numero_visite = 0
        self._spese = 0

    def add_spesa(self, spesa):
        self._numero_visite += 1
        self._spese += spesa

    def spesa_media(self):
        return self._spese / self._numero_visite
